packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -255,8 +257,6 @@ export class DocumentDeltaConnection
      * @returns messages sent during the connection
      */
     public get initialMessages(): ISequencedDocumentMessage[] {
-        // Can't really calling initialMessages() twice - we do not keep ops after first call!
-        assert(this.earlyOpHandler !== undefined, "initialMessages called twice");
```

Comment:
The check here was ensuring that it can be called only once.
assert below only partially covers that case, as "op" handler maybe unregistered.
They other way to say - all this code works correctly only  with a single consumer.
I think we are loosing that here, I'd add a boolean to validate that this method is called only once (and only after "op" handler is setup).

Reply:
Does it matter though?  The `initialMessages` remain the same between calls -- it would be weird to call it twice (maybe reflecting a bug lower in the stack) but it shouldn't have any negative side effects.

Reply:
We stop accumulating ops after making first call. So assuming second user, that second user will miss some ops that arrived in between those too calls on "op" handler. As result, second user would need to hit storage to get those ops. But PUSH (at least in V1) would not flush ops unless new summary posted (or time-out). So client would sit waiting and hitting storage for next 30 seconds at minimum.

Reply:
Basically the contract here - ops are either present in storage, or they are provided through early handler - there can't be a gap here.

Reply:
I'm not sure I follow - the `initialMessages` will still provide those ops and there shouldn't be a gap?  We flush the `queuedMessages` but push them into the `initialMessages` so I believe subsequent callers would still have access to them.

Reply:
Yes, but it will not contain any messages that come on "op" handler after initial call.
The second user who registers "op" handle much later will not see them, neither on "op" handler nor through initialMessages 
The system is not capable of serving second client.

Reply:
Latest change includes a flag and assert to ensure we don't allow this scenario.